### DISCUSSION TITLE
Compiling error

### DISCUSSION THREAD

**Discussion by Aruusi** [shorepine/tulipcc#428] [General]
Hi!
I need help to compile tulip. This moment I get this error:
CMake Error at L:/esp/tulipcc-main/micropython/extmod/extmod.cmake:115 (message): MICROPY_PY_BTREE is enabled but the berkeley-db submodule is not initialised. Run 'make BOARD=N16R8 submodules'

**Comment by dpwe** [comment]
Not trying to be funny, but did you try `make BOARD=N16R8 submodules` ?  That should download the additional packages needed.

**Reply by Aruusi** [reply]
Yes I try.

**Reply by Aruusi** [reply]
I get this and that's why I'm not getting forward:
CMake Error at L:/esp/tulipcc-main/micropython/extmod/extmod.cmake:115 (message):
   MICROPY_PY_BTREE is enabled but the berkeley-db submodule is not initialised.
   Run 'make BOARD=TULIP4_R11 submodules'

**Reply by Aruusi** [reply]
I normally dont use esp-idf only arduino-ide.
